### ISSUE TITLE
Compile time format limited to char

### ISSUE THREAD

**Issue by XZiar** [fmtlib/fmt#922]
I don't know why FMT_STRING is limited to use `char` only since fmt itself support both `char` and `wchar_t`. Is it designed to be that?

There several pieces of code hard coded the char_type:

https://github.com/fmtlib/fmt/blob/3e75ad9822980e41bc591938f26548f24eb88907/include/fmt/format.h#L1257

https://github.com/fmtlib/fmt/blob/3e75ad9822980e41bc591938f26548f24eb88907/include/fmt/format.h#L2273-L2274

https://github.com/fmtlib/fmt/blob/3e75ad9822980e41bc591938f26548f24eb88907/include/fmt/format.h#L3684-L3685

It is easy to solve this problem by adding typedef of char_type in the `struct S`.

**Comment by vitaut** [comment]
> Is it designed to be that?

No, just noone generalized it yet. It is indeed easy to solve and PRs are welcome.
